### PR TITLE
Extend scenario coverage

### LINKED ISSUE #16: Need automated tests
We need automated tests. I'm interested to know whether people prefer Cucumber-based testing as a form of documentation or to keep it close to the metal with RSpec. I'm leaning towards Cucumber (perhaps using Aruba given it's a CLI interface) because mocking out Rake's DSL looks like a nightmare. That's not to say we shouldn't unit test with RSpec where we have nicely isolated classes.

### PR DESCRIPTION
**Pull request by asmith-mdsol**
With the scenarios added here, I think we now have fairly complete test coverage of the current functionality, aside from the overwrite prompting, which I'm going to change up in the near future anyway (with appropriate scenarios added). Consequently, once this is merged I'm going to close #16.

### CODE REVIEW

File: features/support/hooks.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Before do
+  write_file("Rakefile", "require 'dice_bag/tasks'")
```
Comment (jcarres-mdsol) [inline features/support/hooks.rb:2]: is this standard practice? I feel like I would never find this here

File: features/providing_values_per_environment.feature [hunk 2]
```diff
@@ -0,0 +1,35 @@
+Feature: Providing values per environment
+
+  In applications frameworks like Rails it's usual to have configuration against
+  multiple "environments". DiceBag allows you to use a prefixed environment
+  variable to override and existing environment variable, meaning you could
```
Comment (jlo188) [inline features/providing_values_per_environment.feature]: just really minor... but I believe... the `and` should be `an`
(i.e.,  `variable to override an existing environment variable,...`)

Thanks!

File: features/install.feature [hunk 3]
```diff
@@ -0,0 +1,13 @@
+Feature: Install
+
+  In Rails projects, DiceBag tasks are loaded automatically. For other
+  projects, add `require 'dice_bag/tasks'` to your `Rakefile` or wherever your
+  local rake tasks are defined.
+
+  Scenario:
```
Comment (jlo188) [inline features/install.feature]: just wondering, but... should there be a scenario description here?

File: features/support/hooks.rb (hunk 1 above)
Comment (asmith-mdsol) [inline features/support/hooks.rb:2]: I'm covering this in the `install.feature` file and it's described in the top level `README.md`. In future, I'll probably add a `README.md` file in the features directory to point people toward this scenario and a `.nav` file so it appears in the right order in [Relish](https://www.relishapp.com/mdsol/dice-bag/docs).

File: features/providing_values_per_environment.feature (hunk 2 above)
Comment (asmith-mdsol) [inline features/providing_values_per_environment.feature]: You're right, thanks!

File: features/install.feature (hunk 3 above)
Comment (asmith-mdsol) [inline features/install.feature]: Yes, good catch!

File: features/support/hooks.rb (hunk 1 above)
Comment (asmith-mdsol) [inline features/support/hooks.rb:2]: Also, worth noting that I'm pulling [a similar trick as is used in VCR](https://www.relishapp.com/vcr/vcr/v/2-4-0/docs/about-these-examples).

**Comment by jcarres-mdsol** [comment]
Ok, good job, I think this is done and can be merged

**Pull request merged**